src/domain/user.js

```diff
@@ -141,9 +141,9 @@ export default class User {
   static async findById(id, role) {
     const isTeacher = role === 'TEACHER'
     if (isTeacher) {
-      return _findByUniqueAsATeacher('id', id)
+      return _findByUniqueAsATeacher(id, role)
```

Comment:
this should be like the past:
 return _findByUniqueAsATeacher('id', id) -> it is the correct one

Reply:
still needs to correct this one